.gitlab-ci.yml

```diff
@@ -71,18 +70,28 @@ test:acceptance:
     - deb_version=$(cat ${CI_PROJECT_DIR}/output/mender-client-deb-version)
     - python3 -m pytest -v --mender-client-version $MENDER_VERSION --mender-client-deb-version $deb_version
 
-# publish:s3:
-#   stage: publish
-#   image: debian:buster
-#   dependencies:
-#     - build
-#   before_script:
-#     - apt update && apt install -yyq awscli
-#   script:
-#     - echo "Publishing ${CI_COMMIT_REF_NAME} packages to S3"
-#     - aws s3 cp $output/mender-client-master-1_amd64.deb
-#         s3://$S3_BUCKET_NAME/${CI_COMMIT_REF_NAME}/${S3_BUCKET_PATH}/amd64/mender-client-master-1_amd64.deb
-#     - aws s3api put-object-acl --acl public-read --bucket $S3_BUCKET_NAME
-#         --key ${CI_COMMIT_REF_NAME}/${S3_BUCKET_PATH}/amd64/mender-client-master-1_amd64.deb
-#   only:
-#     - /^(master|[0-9]+\.[0-9]+\.x)$/
+publish:s3:
+  stage: publish
+  image: debian:buster
+  dependencies:
+    - build
+  before_script:
+    - apt update && apt install -yyq awscli
+  script:
+    - echo "Publishing ${MENDER_VERSION} packages to S3"
+    - deb_version=$(cat output/mender-client-deb-version)
+    # For master packages, the Debian version is "0.0~[iso-date].[git-hash]-1" and we make a copy named "master" for GUI to use
```

Comment:
Under which circumstances is the GUI using this and not the official package?

Reply:
Maybe I am misunderstanding your question. What do you mean by official package? The official Debian packages? That ones are never used in the GUI, we only serve our own ones.

Then, a master build will make the GUI serve a link of the form `.../master/dist-packages/debian/mender-client_master-1_armhf.deb`, and that is why we make the copy. So in a way this "master" really points to the latest of all the `0.0~git...`

Reply:
By official I mean our own package, but the versioned one. I mean, where do we want to serve the master package and not the latest stable one? The former is not tested by us.

Reply:
@mzedel Please see above for context, as I am not confident on my understanding. Will the GUI link to the mender-client master package under some circumstances?

Reply:
The .deb package `master` will be linked to in the help section and onboarding, unless a different mender version is defined - FWIW it used to just always be master.